functions/go/set-label/main.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kustomize/api/k8sdeps/kunstruct"
+	"sigs.k8s.io/kustomize/api/konfig/builtinpluginconsts"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/yaml"
+)
+
+type transformerConfig struct {
+	FieldSpecs types.FsSlice `json:"commonLabels,omitempty" yaml:"commonLabels,omitempty"`
+}
+
+type functionConfigData struct {
+	LabelName  string `json:"label_name,omitempty" yaml:"label_name,omitempty"`
+	LabelValue string `json:"label_value,omitempty" yaml:"label_value,omitempty"`
+}
+
+//nolint
+func main() {
+	var plugin *plugin = &KustomizePlugin
+	tc, err := getDefaultConfig()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	resmapFactory := newResMapFactory()
+
+	pluginHelpers := newPluginHelpers(resmapFactory)
+
+	resourceList := &framework.ResourceList{}
+	resourceList.FunctionConfig = map[string]interface{}{}
+
+	cmd := framework.Command(resourceList, func() error {
+		resMap, err := resmapFactory.NewResMapFromRNodeSlice(resourceList.Items)
+		if err != nil {
+			return errors.Wrap(err, "failed to convert items to resource map")
+		}
+		data, err := getData(resourceList.FunctionConfig)
+		if err != nil {
+			return errors.Wrap(err, "failed to get data field from function config")
+		}
+		if data.LabelName == "" || data.LabelValue == "" {
+			return fmt.Errorf("label_name and label_value cannot be empty")
+		}
+		dataBytes, err := getDataBytes(resourceList.FunctionConfig)
+		if err != nil {
+			return errors.Wrap(err, "failed to get data bytes")
+		}
+
+		err = plugin.Config(pluginHelpers, dataBytes)
+		if err != nil {
+			return errors.Wrap(err, "failed to config plugin")
+		}
+		// append default field specs
+		plugin.FieldSpecs = append(plugin.FieldSpecs, tc.FieldSpecs...)
+		// set label key and value
+		plugin.Labels = make(map[string]string)
+		plugin.Labels[data.LabelName] = data.LabelValue
+
+		err = plugin.Transform(resMap)
+		if err != nil {
+			return errors.Wrap(err, "failed to run transformer")
+		}
+
+		resourceList.Items, err = resMap.ToRNodeSlice()
+		if err != nil {
+			return errors.Wrap(err, "failed to convert resource map to items")
+		}
+		return nil
+	})
+
+	cmd.Long = usage()
+	if err := cmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func usage() string {
+	return `Add a label to all resources.
+
+Configured using a ConfigMap with the following keys:
+
+label_name: Label name to add to resources.
+label_value: Label value to add to resources.
+
+Example:
+
+To add a label 'color: orange' to all resources:
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  label_name: color
```

Comment:
Do we support adding more than 1 labels at the same time?

Reply:
No, this function only supports one label a time.

Reply:
Comparing with the functionality in kustomize, user can specify multiple labels under `commonLabels` in `kustomization.yaml`. 
When a kustomize user tries to use this function, he/she may expect it to be able to change multiple labels at once.

We need to decide this for set-label and set-annotation. @frankfarzan WDYT?

If we want to do multiple, we may have something like this:
```
type functionConfigData struct {
	Labels  []label `json:"labels,omitempty" yaml:"labels,omitempty"`
}

type label struct {
	LabelName  string `json:"label_name,omitempty" yaml:"label_name,omitempty"`
	LabelValue string `json:"label_value,omitempty" yaml:"label_value,omitempty"`
}
```

Reply:
Updated the PR to support multiple labels.